File: code/geometry/point_in_polygon.py
def pointInPolygon(poly, pt, strict = True):
	c = False
	for i in range(len(poly)):
		q = poly[i - 1]
		if onSegment(q, poly[i], pt):
			return not strict
		c ^= ((pt[1] < q[1]) - (pt[1] < poly[i][1])) * cross(q, poly[i], pt) > 0
	return c

File: code/geometry_cpp/point_in_polygon.cpp
bool pointInPolygon(const vector<ipoint>& poly, ipoint p, bool strict = true) {
	bool c = false;
	auto prev = poly.back();
	for (auto cur : poly) {
		if (onSegment(prev, cur, p)) return !strict;
		c ^= ((p.y < prev.y) - (p.y < cur.y)) * prev.cross(cur, p) > 0;
		prev = cur;
	}
	return c;
}
